mesa/space.py

```diff
@@ -418,36 +419,18 @@ def is_cell_empty(self, pos: Coordinate) -> bool:
         x, y = pos
         return self.grid[x][y] == self.default_val()
 
-    def move_to_empty(
-        self, agent: Agent, cutoff: float = 0.998, num_agents: int | None = None
-    ) -> None:
+    def move_to_empty(self, agent: Agent, cutoff: float = 0.002) -> None:
```

Comment:
While the new cutoff definition is easier to understand (with a number close to epsilon), this breaks backward compatibility, by changing the definition of the `cutoff` argument.

We should defer this breaking change in a separate PR. In that PR, it might be safer to rename the argument to `threshold` instead of cutoff.

Reply:
ok, will open soon

Reply:
Actually, the fact that `num_agents` is removed already makes it backward incompatible. This is a tricky decision to make. I think we should keep this PR as 100% refactor, and so, the `num_agents` argument should remain present, even though it is not going to be used.

Reply:
There should be a warning shown to the user when `num_agents` is not `None`.